src/validation.cpp

```diff
@@ -5204,6 +5209,22 @@ bool ChainstateManager::PopulateAndValidateSnapshot(
     return true;
 }
 
+Chainstate& ChainstateManager::GetChainstateForNewBlock(const uint256& blockhash)
+{
+    AssertLockHeld(::cs_main);
+    const auto* pblock{m_blockman.LookupBlockIndex(blockhash)};
+    // If pblock is null, we haven't seen the header for this block.
+    // Because we expect to have received the headers for the IBD chain
+    // contents before receiving blocks, this means that any block for
+    // which we don't have headers should go in the snapshot chain.
+    if (m_snapshot_chainstate &&
+            (pblock == nullptr || !m_snapshot_chainstate->m_chain.Contains(pblock))) {
```

Comment:
IIUC: maybe clarify here that `m_chain.Contains` searches both the snapshot headers _and_ the IBD headers it's built on top of. When a `pblock` exists but is not contained in the chainstate it's a fork, and it's up to the snapshot chainstate to explore those.